keps/0010-package-manager.md

```diff
@@ -166,12 +167,82 @@ Our official repository is hosted on Google Cloud Storage and following a flat s
 
 We rely on just an HTTP Server, e.g. the out-of-the-box solution that Google Cloud Storage provides, that serves operator `tgz` files and makes them available to users.
 
-The logic for keeping the operators in sync should live in the CLI and is not defined on this KEP . That way the HTTP server only has to serve the index and the Package `tgz` files, without having to implement any business logic. 
+The logic for keeping the operators in sync should live in the CLI and is not defined on this KEP . That way the HTTP server only has to serve the index and the Package `tgz` files, without having to implement any business logic.
 
 The proposed structure is fairly easy to replicate and highly customizable.
 
 Safety when distributing our Packages is another concern. As we continue working on this KEP we will add more details on how we can verify and prevent `Arbitrary software installation`, `Vulnerability to key compromises`, etc. For now using HTTPS to fetch the index and Packages is considered sufficient.
 
+### Index File Specification
+
+`index.yaml` is the base definition of a repository.  It follows the following reference format.
+
+```yaml
+apiVersion: str(desc='version of repository')
+entries: map[string]PackageVersions
+  str(desc='name of operator'):
+  - name:  str(max=253)
+    version: semver()
+    appVersion: str(desc='component controlled version', required=False)
+    home: url(desc='home page, git repo', required=False)
+    sources: array(str, required=False)
+    - str(desc='source code for this operator')
+    description: str(required=False)
+    maintainers: array(Maintainer, required=False)
+    - name: str()
+      email: email()
+    deprecated: bool(desc='reflects if operator is deprecated')    
```

Comment:
maybe the description should rather say what deprecated means for us? Can you still install? I honestly don't know how exactly we interpret this

Reply:
it leaves me wondering if we should do that by GVK versions. don't deprecate at the field level, deprecate at the GVK. that way, we just support multiple GVKs, deprecate a given GVK in a given release, and remove it after that.

Reply:
I'm not sure I follow completely... that said is that a vote to remove deprecated?

Reply:
Yes. My vote is to deprecate the entire "version" of our operator.yaml spec by adding an apiVersion tag to it at some point, rather than deprecate individual fields.